x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/eql/eql.ts

```diff
@@ -165,8 +166,19 @@ export const eqlExecutor = async ({
 
       let newSignals: Array<WrappedFieldsLatest<BaseFieldsLatest>> | undefined;
 
+      // @ts-expect-error shard_failures exists in
+      // elasticsearch response v 8.18.x
+      // TODO: remove ts-expect-error when ES lib version is updated
+      const shardFailures = response.shard_failures;
+
       const { events, sequences } = response.hits;
 
+      if (shardFailures) {
```

Comment:
Chatted on Zoom, but to summarize - `shardFailures` is returned for both sequence and non-sequence queries, so we can't rely on this alone to determine what error message to display.

Testing it out, it looks great for non-sequence queries, but the same error displays for sequence queries. Still super helpful to show the shard failures for both types of queries, but for sequence queries we'll want to tweak the language just a bit to make it clear that no search was done, even on available shards.